### PR TITLE
prov/rxm: Fix resource leak

### PR DESCRIPTION
**Pull request by gladkovdmitry17**
This patch fixes resource leak if the `fi_getname` for an underlying MSG EP fails:
```
$ FI_LOG_LEVEL=warn I_MPI_OFI_PROVIDER="sockets;ofi_rxm" I_MPI_FABRICS=ofi mpirun -n 2 –ppn 1 manyget

[0] MPI startup(): ofi fabric is not available and fallback fabric is not enabled
libfabric:ofi_rxm:ep_ctrl:rxm_conn_cmap_alloc():380<warn> Unable to fi_getname on msg_ep
libfabric:ofi_rxm:av:ofi_av_close():438<warn> AV is busy
[1] MPI startup(): ofi fabric is not available and fallback fabric is not enabled
libfabric:ofi_rxm:ep_ctrl:rxm_conn_cmap_alloc():380<warn> Unable to fi_getname on msg_ep
libfabric:ofi_rxm:av:ofi_av_close():438<warn> AV is busy
```

After applying this patch:
```
$ FI_LOG_LEVEL=warn I_MPI_OFI_PROVIDER="sockets;ofi_rxm" I_MPI_FABRICS=ofi mpirun -n 2 –ppn 1 manyget
libfabric:ofi_rxm:ep_ctrl:rxm_conn_cmap_alloc():385<warn> Unable to fi_getname on msg_ep
libfabric:ofi_rxm:ep_ctrl:rxm_conn_cmap_alloc():385<warn> Unable to fi_getname on msg_ep
[0] MPI startup(): ofi fabric is not available and fallback fabric is not enabled
[1] MPI startup(): ofi fabric is not available and fallback fabric is not enabled

```

Signed-off-by: Dmitry Gladkov <dmitry.gladkov@intel.com>

### CODE REVIEW

**Comment by gladkovdmitry17** [comment]
This PR is correlated with #3452 
The #3452 fixes the problem where unable to clsoe the AV, because the AV is bound to EP.  I've dropped the changes from this PR (#3450) to avoid merge conflicts

File: prov/rxm/src/rxm_ep.c [hunk 1]
```diff
@@ -1138,6 +1135,13 @@ static int rxm_ep_bind(struct fid *ep_fid, struct fid *bfid, uint64_t flags)
 			if (ret)
 				return ret;
 		}
+		ret = ofi_ep_bind_cq(&rxm_ep->util_ep, cq, flags);
+		if (ret) {
+			if (cq->wait && ofi_wait_fd_del(cq->wait, rxm_ep->msg_cq_fd))
+				FI_INFO(&rxm_prov, FI_LOG_EP_CTRL,
+					"Unable to delete wait fd from FD list");
+			return ret;
+		}
```
Comment (shefty) [inline prov/rxm/src/rxm_ep.c]: Is this change related to #3452?  Or is this part of the resource leak cleanup?

Comment (gladkovdmitry17) [reply prov/rxm/src/rxm_ep.c]: It's part of the resource leak cleanup.
Before I've upstremed #3452, I had changes to the AV binding to the EP. I've dropped them

File: prov/rxm/src/rxm_conn.c [hunk 2]
```diff
@@ -370,7 +370,12 @@ struct util_cmap *rxm_conn_cmap_alloc(struct rxm_ep *rxm_ep)
 	int ret;
 
 	len = rxm_ep->msg_info->src_addrlen;
-	name = malloc(len);
+	name = calloc(1, len);
```
Comment (fzago-cray) [inline prov/rxm/src/rxm_conn.c]: Why the change? Other caller don;'t zero out the buffer.

Comment (gladkovdmitry17) [reply prov/rxm/src/rxm_conn.c]: If the underlying provider fills not entire buffer, we should know the boundaries of the buffer that was filled.
Yes, I know we have a length. But I guess it would add some sort of safety

Comment (fzago-cray) [reply prov/rxm/src/rxm_conn.c]: It doesn't matter since rxm_conn_cmap_alloc is memdup'ing src_addrlen bytes from that buffer.

And that also means that name is leaked when rxm_conn_cmap_alloc exits.

Comment (gladkovdmitry17) [reply prov/rxm/src/rxm_conn.c]: @fzago-cray Thanks for finding this. It's really memory leak. I'll fix this!

Comment (gladkovdmitry17) [reply prov/rxm/src/rxm_conn.c]: I've updated to fix memory leak

Comment (fzago-cray) [reply prov/rxm/src/rxm_conn.c]: Looks good. Thanks.

Comment (gladkovdmitry17) [reply prov/rxm/src/rxm_conn.c]: @fzago-cray Thank you too! it was a good catch!
